src/service/gnuradio/GnuRadioWorker.hpp

```diff
@@ -202,35 +203,32 @@ class GnuRadioAcquisitionWorker : public Worker<serviceName, TimeDomainContext,
-                    // TODO using a multithreaded scheduler requires some fixes (frank/scheduler_fixes branch and the per-block lifecycle methods (stop(), in particular)), so we use a
-                    // single-threaded scheduler for now until those issues are solved. This prevents terminating the running scheduler and replacing the graph.
-#if 0
+                    auto runScheduler = [&stopScheduler, &schedulerFinished](gr::Graph &&fg) {
                         using Scheduler = gr::scheduler::Simple<gr::scheduler::ExecutionPolicy::multiThreaded>;
-                        // TODO currently we limit ourselves to one thread here, because
-                        // - (seen with 2 threads) the data sink receives too large chunks, e.g. samples 0..100 when there's a tag at 0 and 50, instead of 0..49 (see triggered test case)
-                        // - (seen with unbound thread count) the qa_DataSink test (graph-prototype) fails completely when using a multithreaded scheduler with arbitrary number of threads
-                        auto threadPool = std::make_shared<thread_pool::BasicThreadPool>("simple-scheduler-pool", thread_pool::CPU_BOUND, 1UZ, 1UZ);
-                        auto scheduler  = Scheduler(std::move(fg), threadPool);
+                        auto scheduler  = Scheduler(std::move(fg));
+                        scheduler.init();
                         scheduler.start();
                         while (!stopScheduler) {
                             std::this_thread::sleep_for(std::chrono::milliseconds(10));
                             if (!scheduler.isProcessing()) {
                                 schedulerFinished = true;
                             }
                         }
+
                         scheduler.stop();
-#else
-                        gr::scheduler::Simple sched(std::move(fg));
-                        sched.runAndWait();
-                        schedulerFinished = true;
-#endif
                     };
 
                     stopScheduler     = false;
                     schedulerFinished = false;
-                    handleSubscriptions(streamingPollers, dataSetPollers); // create pollers for new sinks before starting graph
-                    schedulerThread = std::jthread(runScheduler, std::move(*pendingFlowGraph));
+                    try {
+                        auto graph = gr::load_grc(*_plugin_loader, *pendingFlowGraph);
+                        handleSubscriptions(streamingPollers, dataSetPollers); // create pollers for new sinks before starting graph
+                        schedulerThread = std::jthread(runScheduler, std::move(graph));
+                    } catch (...) {
+                        // must never happen, parsing ensured by FlowGraphWorker
+                        fmt::println(std::cerr, "Internal error: Could not parse flow graph: {}", *pendingFlowGraph);
+                        std::terminate();
```

Comment:
Ok for now, since it is checked elsewhere but ideally the service should not be able to terminate when there's a way to recover which I guess here should be possible. To be revisitied.